lib/elixir_exif.ex

```diff
@@ -130,4 +119,63 @@ defmodule ElixirExif do
   defp get_field_byte_length(@undefined, component_count), do: component_count
   defp get_field_byte_length(@slong, component_count), do: 4*component_count
   defp get_field_byte_length(@srational, component_count), do: 8*component_count
+
+  defp parse_first_ifds(fields, app1, offset, read_unsigned) do
+    {fields, next_offset} = parse_idf(:tiff, app1, offset, read_unsigned, fields) # parse 0th IFD
+
+    if next_offset == 0 do
+      fields
+    else
+      {new_fields, _} = parse_idf(:tiff, app1, next_offset, read_unsigned, fields) # parse 1st IFD
+      new_fields
+    end
+  end
+
+  defp parse_exif_ifd(fields, app1, read_unsigned) do
```

Comment:
These three parse functions have a lot of duplication in them, I think it would be better for them to call a common function similar to
```elixir
@spec parse_idf_fields(id :: integer, type :: :exif|:gps|:interop, fields, app1, read_unsigned)
```